apps/finance/app/src/components/Transfers.js

```diff
@@ -161,143 +158,184 @@ const Transfers = React.memo(({ dao, tokens, transactions }) => {
-              <a role="button" onClick={handleResetFilters}>
-                Clear filters
-              </a>
-            )}
-          </p>
-        </NoTransfers>
-      ) : (
-        <div>
-          <Table
-            compactMode={compactMode}
-            header={
-              !compactMode && (
-                <TableRow>
-                  <TableHeader title="Date" css="width: 12%" />
-                  <TableHeader title="Source / Recipient" css="width: 40%" />
-                  <TableHeader title="Reference" css="width: 100%" />
-                  <TableHeader title="Amount" align="right" css="width: 0" />
-                  <TableHeader />
-                </TableRow>
-              )
-            }
+    <DataView
+      page={page}
+      onPageChange={setPage}
+      heading={
+        <React.Fragment>
+          <div
             css={`
-              color: ${theme.textPrimary};
-              margin-bottom: 20px;
+              padding: ${2 * GU}px 0;
+              display: flex;
+              align-items: center;
+              justify-content: space-between;
             `}
           >
-            {filteredTransfers
-              .sort(({ date: dateLeft }, { date: dateRight }) =>
-                // Sort by date descending
-                compareDesc(dateLeft, dateRight)
-              )
-              .slice(0, displayedTransfers)
-              .map(transfer => (
-                <TransferRow
-                  key={transfer.transactionHash}
-                  token={tokenDetails[toChecksumAddress(transfer.token)]}
-                  transaction={transfer}
-                  smallViewMode={compactMode}
-                />
-              ))}
-          </Table>
-          {displayedTransfers < filteredTransfers.length && (
-            <Footer compactMode={compactMode}>
-              <Button mode="secondary" onClick={showMoreTransfers}>
-                Show Older Transfers
+            <div
+              css={`
+                color: ${theme.content};
+                ${textStyle('body1')}
+              `}
+            >
+              Transfers
+            </div>
+            <div css="text-align: right;">
+              <Button onClick={handleDownload}>
+                <IconExternal /> Export
               </Button>
-            </Footer>
+            </div>
+          </div>
+          {layoutName !== 'small' && (
+            <TransfersFilters
+              dateRangeFilter={selectedDateRange}
+              onDateRangeChange={handleSelectedDateRangeChange}
+              tokenFilter={selectedToken}
+              onTokenChange={handleTokenChange}
+              transferTypeFilter={selectedTransferType}
+              onTransferTypeChange={handleTransferTypeChange}
+              compactMode={compactMode}
+              symbols={symbols}
+              transferTypes={TRANSFER_TYPES_STRING}
+            />
           )}
-        </div>
+          {emptyResultsViaFilters && (
```

Comment:
cc @bpierre the location of this (in the header, to avoid the table headers) is kind of surprising. Maybe we should add an "empty" mode to `DataView`?

Reply:
I don't think there is a way to hide the headers currently (please confirm @bpierre ) so I added this quick fix but this should definitely be refactored.

Reply:
Yes! It could be used for the filters, when there is no data at all, and when the data is loading (we might want to separate this state though).

@owisixseven @dizzypaty What would you think of providing the default loading state and empty states for DataView?

We already have the “no results for these filters”:

![image](https://user-images.githubusercontent.com/36158/62057189-72b97d80-b21f-11e9-9eeb-60485cb35faa.png)

And we could add two variants: “no data” and “loading data”. It would only be the default, we could still let app authors change the illustration, the content or even everything if needed.

Reply:
Oops sorry @AquiGorka GitHub didn’t show your message until I posted mine.

> I don't think there is a way to hide the headers currently (please confirm @bpierre ) so I added this quick fix but this should definitely be refactored.

Ah no we can’t hide them… an immediate solution would be to render the filters in an empty state component when there are no results, but I think we should provide this empty state screen from DataView directly, as discussed above.